src/network.rs

```diff
@@ -123,10 +123,37 @@ pub enum Event {
 
 #[derive(Debug)]
 pub struct SwapMetadata {
-    shared_swap_id: SharedSwapId,
+    swap_id: SwapId,
     taker_peer_id: PeerId,
 }
 
+impl SwapMetadata {
+    pub fn swap_id(&self) -> SwapId {
+        self.swap_id
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct SwapId {
```

Comment:
🙃 As far as I can see we are still just using the `SharedSwapId`, not sure if we have to care about the "inner" one.
Or did I miss something?

Reply:
We use the inner one to derive Bitcoin transient keys from it.